src/main/java/com/google/api/codegen/viewmodel/OutputView.java

```diff
@@ -0,0 +1,85 @@
+/* Copyright 2018 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.api.codegen.viewmodel;
+
+import com.google.auto.value.AutoValue;
+import com.google.common.collect.ImmutableList;
+
+@AutoValue
+public abstract class OutputView {
+
+  public enum Kind {
+    PRINT
+  }
+
+  public abstract Kind kind();
+
+  public abstract PrintView print();
```

Comment:
Are you thinking of doing essentially a oneof pattern here as we add more `Kind`s? How about making `OutputView` abstract and making `PrintView`, `LoopView`, etc be subclasses? That way each `OutputView` is guaranteed to be exactly one of those, and the`MethodSampleView` has the top-level collection of such `OutputView`s.

Reply:
No, we can  still have a `kind()` method in`OutputView` that gets overriden by subclasses. Wouldn't that make the `.snip` side of things just as easy as it is now?

Reply:
Done